### ISSUE TITLE
Modularity

### ISSUE THREAD

**Issue by Yomguithereal** [simple-statistics/simple-statistics#180]
It would be nice to be able to require functions one by one likewise:

```js
var mean = require('simple-statistics/mean');
// Now, you can kinda do this by writing:
var mean = require('simple-statistics/src/mean');
// This works but feel hacky and will break if you change folder structure
```

But I know this is cumbersome because this means either to move every file into root folder or to have a build process copy files etc. Just a wild idea.

**Comment by tmcw** [comment]
The latter is what I've documented in the past, and I think for CommonJS modules, it's roughly equivalent to the former: if we moved `mean` into the root folder, then moving it anywhere else would break, just the same as moving `src` anywhere else. I don't really see one or the other being more 'maintainable' because regardless of where you put the files, you're committing to keeping them there.

I think for the CommonJS approach what we have now is roughly the correct approach - doesn't require a build step, incurs a minor typing penalty (`src`).

ES6 modules with code splitting would make this maintainable by keeping file locations an internal detail, but my experiments with them have so far been disappointing for this module - they'd require a build step, extra testing code, and they'd break the CommonJS approach. And while Rollup does a pretty robust job of code splitting, webpack's implementation still has a lot of warts that prevents it from fully working in many cases, like [when you have multiple entry points](https://github.com/webpack/webpack/issues/2855).
